tests/jballerina-unit-test/src/test/resources/test-src/services/service_attach_test.bal

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+isolated boolean[] nameParamCheck = [false, false, false, false];
+
+public class Listener1 {
+    public isolated function 'start() returns error? { }
+
+    public isolated function gracefulStop() returns error? { }
+
+    public isolated function immediateStop() returns error? { }
+
+    public isolated function detach(service object {} s) returns error? { }
+
+    public isolated function attach(service object {} s, string? name) returns error? {
+        lock {
+            nameParamCheck[0] = name is () && !(<any> name is string[]);
+        }
+    }
+}
+
+service on new Listener1() {
+
+}
+
+public class Listener2 {
+    public isolated function 'start() returns error? { }
+
+    public isolated function gracefulStop() returns error? { }
+
+    public isolated function immediateStop() returns error? { }
+
+    public isolated function detach(service object {} s) returns error? { }
+
+    public isolated function attach(service object {} s, string? name) returns error? {
+        lock {
+            nameParamCheck[1] = name is string && name == "Service";
+        }
+    }
+}
+
+service "Service" on new Listener2() {
+
+}
+
+public class Listener3 {
+    public isolated function 'start() returns error? { }
+
+    public isolated function gracefulStop() returns error? { }
+
+    public isolated function immediateStop() returns error? { }
+
+    public isolated function detach(service object {} s) returns error? { }
+
+    public isolated function attach(service object {} s, string[]? name) returns error? {
+        lock {
+            nameParamCheck[2] = name is string[] && name[0] == "foo" && name[1] == "bar" && name.length() == 2;
+        }
+    }
+}
+
+service /foo/bar on new Listener3() {
+
+}
+
+public class Listener4 {
+    public isolated function 'start() returns error? { }
+
+    public isolated function gracefulStop() returns error? { }
+
+    public isolated function immediateStop() returns error? { }
+
+    public isolated function detach(service object {} s) returns error? { }
+
+    public function attach(service object {} s, string[]? name) returns error? {
+            lock {
+                nameParamCheck[3] = name is string[] && name[0] == "/" && name.length() == 1;
```

Comment:
The spec says
> The attach-point determines the second argument passed to the attach method: if the attach-point is absent, then the argument is nil; if it is a string-literal, then the argument is a string; otherwise, **it is an absolute-resource-path and the argument is an array of strings, with one string for each resource-path-segment.**

In this case, is it correct for the array to have a single member "/" or should it be an empty array? :thinking:

Reply:
Spec issue created as from I can remember, it was required to pass a string array with a single `/` member.
https://github.com/ballerina-platform/ballerina-spec/issues/810

Reply:
Issue created to fix this spec deviation https://github.com/ballerina-platform/ballerina-lang/issues/29960

Reply:
I think we have to remove the `if` block in line:209-212 of desugar class and align test cases with the change

Reply:
@rdhananjaya I did the necessary changes. But if we removed lines 209-212, we will not be handling the nil literal case right? Anyway did the changes, please review

Reply:
Oh yes you are correct, my bad.